src/mbgl/layout/symbol_layout.cpp

```diff
@@ -90,10 +92,10 @@ SymbolLayout::SymbolLayout(std::string bucketName_,
                 u8string = platform::lowercase(u8string);
             }
 
-            ft.text = util::utf8_to_utf32::convert(u8string);
+            ft.text = bidi.bidiTransform( util::utf8_to_utf16::convert( u8string ) );
```

Comment:
I pushed my changes so you can take a look at how I did the line breaking, but the results for the Library of Alexandria are pretty disheartening.